### PR TITLE
sql runner: graduate out of a proof of concept

### PR DESCRIPTION
**Pull request by mping-exo**
This patch brings the necessary missing functionality for production
use of the SQL runner, namely:
    
- Full serialization support of AST nodes, including augment stanzas
  This introduces a breaking change, preventing functions from being
  used as augment sources. Instead, the `'identity` symbol is now a valid
  source to produce the full output at a specific path in the augment
  destination. **THIS IS A BREAKING CHANGE**.
- Relevant indices have been added to the `workflow_run` and `task`
  tables.
- Tests run on both the manifold runner and SQL runner.
- New fields have been added to both `workflow_run` and to `task` to
  serve common request patterns more efficiently.
- No support for the runtime field, the `action-fn` being used should close
  over its dependencies

### CODE REVIEW

File: src/ablauf/job.clj [hunk 1]
```diff
@@ -66,9 +66,10 @@
       (= :result/success result)
       (assoc-in dest-vec
                 (cond
-                  (sequential? source) (get-in output source)
-                  (keyword? source)    (get output source)
-                  (fn? source)         (source output))))))
+                  (sequential? source)      (get-in output source)
+                  (= ::ast/identity source) output
```
Comment (mping-exo) [inline src/ablauf/job.clj:70]: Breaking change to support (de)serialization of nodes with augmentation

**Comment by mping-exo** [comment]
@pyr @tdantas some notes:

- tests are passing with both manifold and sql runner

- created a macro to run parameterized tests: `(deftestp expected-failure-test [arunner [runner sql-runner]]`
	- expands to something like
```
(doseq [arunner [runner sql-runner]]
  (testing (str "arunner:"arunner) ....))
```

- deal with `runtime` on job restart
  -  I decided to ditch saving the jobid on the runtime to ensure tests pass
  - the `exec/context` wasnt being propagated, same for runtime
- serialize `(with-augment [identity :out])`
	-  experimented with using vars `#identity` and custom edn readers, opted by using `::ast/identity` as it seems our codebase only uses `identity` as augment src, so we  would need to migrate those jobs to use `[::ast/identity ...]` instead

We should need to deal with:
- failure of the main worker `loop`: it will hang the worker thread and no job will finish
- some kind of logging to ensure minimal visibility in case everything fails

File: src/ablauf/job/ast.clj [hunk 2]
```diff
@@ -124,6 +124,8 @@
   (get (:ast/nodes node) 2))
 
 (defn with-augment
-  "Provide a source and dest for `bundes.job/augment`"
+  "Provide a source and dest for ast augment"
   [[source dest] node]
+  (when (fn? source)
+    (throw (IllegalArgumentException. "Cannot use functions as augment source; consider using ::ast/identity for identity")))
```
Comment (tgrondier) [inline src/ablauf/job/ast.clj:130]: Note for future prs: it would be nice to add proper documentation to `with-augment` while we're at it

Comment (pyr) [reply src/ablauf/job/ast.clj]: @tgrondier: the docstring should point to `ablauf.job/augment` which explains the whole idea.

**Review by wdullaer — APPROVED** [review]

**Pull request merged**